system_tests/common_test.go

```diff
@@ -182,25 +182,31 @@ func (l *lifecycle) Stop() error {
 }
 
 func createTestL1BlockChain(t *testing.T, l1info info) (info, *ethclient.Client, *eth.Ethereum, *node.Node) {
+	return createTestL1BlockChainWithConfig(t, l1info, nil)
+}
+
+func createTestL1BlockChainWithConfig(t *testing.T, l1info info, stackConfig *node.Config) (info, *ethclient.Client, *eth.Ethereum, *node.Node) {
 	if l1info == nil {
 		l1info = NewL1TestInfo(t)
 	}
+	if stackConfig == nil {
+		defaultStackConfig := node.DefaultConfig
+		stackConfig = &defaultStackConfig
+		stackConfig.HTTPPort = 0
+		stackConfig.WSPort = 0
+		stackConfig.UseLightweightKDF = true
+		stackConfig.P2P.ListenAddr = ""
+		stackConfig.P2P.NoDial = true
+		stackConfig.P2P.NoDiscovery = true
+		stackConfig.P2P.NAT = nil
+		stackConfig.DataDir = t.TempDir()
```

Comment:
Could we make a function for this code to make a default stack config that could be used as the base for the ipc test too?